sugar/chains.py

```diff
@@ -208,12 +227,103 @@ async def process_batch(batch: List[Tuple]):
                 for offset, limit in batch: batcher.add(f(limit, offset))
                 return sum(await batcher.async_execute(), [])
         return sum(await asyncio.gather(*[process_batch(batch) for batch in self.get_pool_paginator()]), [])
+    
+    @require_async_context
+    async def get_domain(self, chain_id: Optional[int] = None) -> int:
+        # TODO: remove chain_id arg when all chains support domains
+        # TODO: move this somewhere else
+        MESSAGE_MODULE_ADDRESS = normalize_address("0x2BbA7515F7cF114B45186274981888D8C2fBA15E")
+        domains_abi = [
+            {
+                "name": "domains",
+                "type": "function",
+                "stateMutability": "view",
+                "inputs": [
+                    {
+                        "name": "",
+                        "type": "uint256"
+                    }
+                ],
+                "outputs": [
+                    {
+                        "name": "domain",
+                        "type": "uint256"
+                    }
+                ]
+            }
+        ]
+        contract = self.web3.eth.contract(address=MESSAGE_MODULE_ADDRESS, abi=domains_abi)
+        domain = await contract.functions.domains(int(self.settings.chain_id) if not chain_id else chain_id).call()
+        # TODO: remove fallback to chain_id when all chains support domains
+        return domain if domain != 0 else int(chain_id if chain_id else self.settings.chain_id)
+    
+    @require_async_context
+    async def get_remote_interchain_account(self, destination_domain: int):
+        abi = [{
+            "name": "getRemoteInterchainAccount",
+            "type": "function",
+            "stateMutability": "view",
+            "inputs": [
+                {
+                    "name": "",
+                    "type": "uint32"
+                },
+                {
+                    "name": "",
+                    "type": "address"
+                },
+                {
+                    "name": "",
+                    "type": "bytes32"
+                }
+            ],
+            "outputs": [
+                {
+                "name": "userICA",
+                "type": "address"
+                }
+            ]
+        }]
+        contract = self.web3.eth.contract(address=self.settings.interchain_account_addr, abi=abi)
+        return await contract.functions.getRemoteInterchainAccount(
+            destination_domain,
+            self.settings.swapper_contract_addr,
+            # always pass user address for consistency. Fallback condition here just to prevent runtime error
+            to_bytes32(self.account.address or ADDRESS_ZERO),
```

Comment:
Thinking we should drop the fallback though, or validate the address is non-zero earlier, wdyt?

Reply:
tbh, just copy-pasted this from the original implementation. going to update. fail early, always ✊